### PR TITLE
Update task rendering documentation to indicate support for all Musta…

### PR DESCRIPTION
**Pull request by benbp**
…che features

Adding documentation to match the features added in with https://github.com/RackHD/on-tasks/pull/76

@RackHD/corecommitters @WangWinson @VulpesArtificem @heckj

### CODE REVIEW

File: docs/rackhd/tasks.rst [hunk 1]
```diff
@@ -208,7 +208,7 @@ Task Templates
 ^^^^^^^^^^^^^^^^^^^^^^^
 There are some values that may be needed in a task definition which are not known in advance. In some cases, it is also more convenient to use placeholder values in a task definition than literal values. In these cases, a simple template rendering syntax can be used in task definitions. Rendering is also useful in places where two or more tasks need to use the same value (e.g. options.file), but it cannot be hardcoded ahead of time.
 
-Task templates use a mustache-style syntax. To define a value to be rendered, place it within curly braces in a string:
+Task templates use mustache syntax (http://mustache.github.io/mustache.5.html), with some additional features detailed below. To define a value to be rendered, place it within curly braces in a string:
```
Comment (heckj) [inline docs/rackhd/tasks.rst]: could you make this a RST hyperlink instead:

```
   `mustache syntax`_
```
and
```
    .. _mustache syntax: http://mustache.github.io/mustache.5.html
```

File: docs/rackhd/tasks.rst [hunk 2]
```diff
@@ -290,6 +290,10 @@ On creation, the options are rendered as below. The 'file' field is specified in
 Task Rendering Features
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+For a full list of Mustache rendering features, including specifying conditionals and iterators, see the `Mustache man page <http://mustache.github.io/mustache.5.html>`_
```
Comment (heckj) [inline docs/rackhd/tasks.rst]: hyperlink here as well?

**Comment by heckj** [comment]
nits around hyperlink formatting, otherwise :+1:

File: docs/rackhd/tasks.rst (hunk 2 above)
Comment (benbp) [inline docs/rackhd/tasks.rst]: Isn't this an RST hyperlink syntax as well? I'm happy to normalize though.

**Comment by heckj** [comment]
:+1:

**Comment by benbp** [comment]
I'll let @WangWinson review in case he has any corrections, then I'll merge.

**Comment by yyscamper** [comment]
the additional feature should speak of the delay render or runtime render (you may have a better name) for the key "context", if template starts with context (like context.value) will be rendered during workflow runtime after task is scheduled, other template rendering occurs immediately when the task instance is created.

Also, I want to mention that the contiomal or loop syntax has not been enable for context rendering, this need enhancement.

File: docs/rackhd/skus.rst [hunk 3]
```diff
@@ -270,13 +270,13 @@ package may have a static and template directory.
...
-    templates/
-    templates/ansible.pub
-    templates/esx-ks
+    config.json
+    static/
+    static/common/
+    static/common/discovery.overlay.cpio.gz
+    templates/
+    templates/ansible.pub
+    templates/esx-ks
```
Comment (WangWinson) [inline docs/rackhd/skus.rst]: What happens to this file ? Looks like nothing changed. Or something wrong with the diff ?

**Comment by WangWinson** [comment]
Thanks @benbp  for update document. Looks good to me :+1:

**Comment by yyscamper** [comment]
:+1: better to speak of the "context" rendering, but this doc is good to me.

**Pull request merged**